### PR TITLE
Fedora nightly scenario

### PR DESCRIPTION
**Pull request by jelly**
In general the idea is:

* Create a new Github workflow which runs nightly
* Github workflow invokes `./bots/tests-trigger --repo cockpit-project/cockpit-podman "-" $DEFAULT_OS/nightly`
* Commit status is posted, webhook runs and the we end up running: `/tests-scan -s $sha --context $DEFAULT_OS/nightly --repo cockpit-project/cockpit-podman`
* Issue is created if fails, otherwise :heavy_check_mark: should be shown on the commit.

Open discussion points:

* [ ] It seems we can't get the direct logs url from s3-streamer, we can either pol statuses or set an `env` variable so `s3-streamer`'s children know where to look @martinpitt what is your opinion here?
* [ ] Another discussion point, do we drop `--pull-number 0` and add a new argument to `tests-invoke`.

Testing ideas:

We could test the new `tests-invoke` option if we:
* [ ] Provide a mock Github issues endpoint where we can create an issue
* [ ] mock `BASE_DIR` so we can control the executed `test/run` and simply `exit 1` or `exit 0`
* [ ] mock / override `time.sleep(60)` or change the polling to be more aggressive as waiting 1 minute in a test is bonkers.

### CODE REVIEW

**Comment by jelly** [comment]
`tests-invoke` is sadly not easy to test without waiting ages on `time.sleep(60)`. Or'd we have to hacky invoke `main()`, but I'll skip that for now.

File: tests-invoke
Comment (martinpitt) [inline tests-invoke]: Yeah, that still needs to happen :)

File: tests-scan
Comment (martinpitt) [inline tests-scan:301]: perhaps change the comment to "projects which have a "master" branch, to clarify what it means? But this is actually easy to do, testmap.py has a `get_default_branch(repo)` function.

File: tests-invoke
Comment (martinpitt) [inline tests-invoke:96]: Ah, please squash this.

File: tests-invoke
Comment (martinpitt) [inline tests-invoke:97]: Any chance this could still be `if not opts.pull_number`, so that it could be `None`? That would be cleaner than `"0"`.

**Review by martinpitt — CHANGES_REQUESTED** [review]
Cheers! Could you please add a unit test for the `tests-scan` part of this, using your shiny new test-tests-scan?

**Comment by jelly** [comment]
Pushed with a test commit which tests the issue creation, but this leads to 60 second delay in tests due to https://github.com/cockpit-project/bots/blob/main/tests-invoke#L94

File: task/test-tests-scan
Comment (martinpitt) [inline task/test-tests-scan:236]: Could this check the content of the issue? that the expected SHA made it into the body? Or doesn't our mock server go that far?

**Review by martinpitt — DISMISSED** [review]
This looks really cool, and I don't want to block this for any longer. Thanks!

I suggest to change the sleep to something like `int(os.getenv("TEST_INVOKE_SLEEP", "60"))` and set the env variable to "1" in the unit test. I leave it up to you if you want to do this here or in a follow-up.

File: tests-scan
Comment (martinpitt) [inline tests-scan:307]: Ah, does this also need to change to get_default_branch()? If that is a github label, I'm not sure why it would even be called "main", instead of the "nightly" one you defined in tests-invoke?

**Comment by jelly** [comment]
> This looks really cool, and I don't want to block this for any longer. Thanks!
> 
> I suggest to change the sleep to something like `int(os.getenv("TEST_INVOKE_SLEEP", "60"))` and set the env variable to "1" in the unit test. I leave it up to you if you want to do this here or in a follow-up.

I'll do it here, I do actually **really** dislike such hacks but there isn't a better way :(

**Pull request merged**
